src/components/events/EditModal.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { XIcon } from "../icons/Icons";
+import { ErrorText } from "../form/ErrorText";
+import { LoadingAnimation } from "../icons/LoadingAnimation";
+import { DAYS_ENUM } from "../../utils/common";
+import { GenericInput } from "../form/GenericInput";
+import { IdIcon } from "../icons/Icons";
+import { useUserState } from "../../utils/userState";
+import { DeleteDeal } from "../../server/deal";
+export default function EditModal({ eventData, onDismiss }) {
+    const [user] = useUserState();
+    const [loading, setLoading] = React.useState(false);
+    const { placeId,
+        dayOfWeek,
+        startTime,
+        endTime,
+        title,
+        deals,
+        place,
+        verifiedAt,
+        _id,
+        tags } = eventData
+    const [newDeal, setNewDeal] = useState({})
+    const [error, setError] = useState('')
+    const onClickEditDeal = async () => {
+        try {
+            if(!newDeal) {
+                setError('Please fill fields.')
+                return
+            }
+            await axios.put(`${process.env.REACT_APP_API_URL}/api/deal/edit/${_id}`, {
+                ...newDeal
+            }, {
+                headers: {
+                  Authorization: `${user.token}`,
+                },
+              })
+              window.location.reload()
```

Comment:
nit: I feel like rather than reloading we should do something where it says like a success toast and then closes the modal... Reloading for every single edit makes me feel weird where then I'm back up to the top of the page or something and could get annoying for the user.

Reply:
Yeah, I jsut copied what we did for the delete/approve route, but I like the toast idea a lot